pkg/kube/kclient/client.go

```diff
@@ -123,7 +123,11 @@ func (n *informerClient[T]) AddEventHandler(h cache.ResourceEventHandler) {
 		},
 		Handler: h,
 	}
-	reg, _ := n.informer.AddEventHandler(fh)
+	reg, err := n.informer.AddEventHandler(fh)
+	if err != nil {
+		// Should only happen if its already stopped. We should exit early.
```

Comment:
add a log

Reply:
TBH I am not sure about this. I worry it will confuse users, and there isn't really anything for them to do in this case?

Reply:
What scenario does this happen in? Could be a useful debug log at the very least.

Reply:
Any controller that states after the serialized startup sequence can hit this if the controller starts concurrently with things shutting down. Winning leader election, for example, can cause this -- as does krt